Source/GSICUString.h

```diff
@@ -80,13 +80,13 @@ static inline void free_string(unichar **buf)
  *
  * Buffers created in this way are exception safe when using native exceptions.
  */
-#define TEMP_BUFFER(name, size)\
+#define TEMP_BUFFER(name, length)\
   __attribute__((cleanup(free_string))) unichar *name ##_onheap = 0;\
-  unichar name ## _onstack[64 / sizeof(unichar)];\
+  unichar name ## _onstack[64];\
   unichar *name = name ## _onstack;\
-  if (size > 64)\
+  if (length * sizeof(unichar) > 64)\
```

Comment:
I don't understand why you multiply here. Now the code expects to have the number of unichars in length and only if that is greater than 64 a dynamic allocated buffer is needed.

Reply:
I think `length * sizeof(unichar)` is the actual number of bytes we need for the buffer. Since the stack buffer is 64 bytes, we cannot use it if we need more than 64 bytes (i.e. if length is more than 32).

Or do I have a "Denkfehler" as we say in German? :)

Reply:
The size of the buffer is in unichar, that is the buffer may hold 64 unichar elements. Only when allocating the dynamic buffer we need to switch over to provide the size in bytes (or use calloc).

Ich denke es ist ein Denkfehler :-)

Reply:
Ah yes, thank you for pointing out my Denkfehler! I didn’t consider the type of the "onstack" array.